rumqttc/src/websockets.rs

```diff
@@ -8,6 +10,33 @@ pub enum UrlError {
     Parse(#[from] http::uri::InvalidUri),
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum ValidationError {
+    #[error("Websocket response does not contain subprotocol header")]
+    SubprotocolHeaderMissing,
+    #[error("Websocket subprotocol header: {0}")]
+    SubprotocolMqtt(String),
+}
+
+pub(crate) fn validate_response_headers(
+    response: Response<Option<Vec<u8>>>,
+) -> Result<(), ValidationError> {
+    let val = response
+        .headers()
+        .get("Sec-WebSocket-Protocol")
+        .ok_or(ValidationError::SubprotocolHeaderMissing)?;
+
+    let sub_protocol = val
+        .to_str()
+        .map_err(|_| ValidationError::SubprotocolHeaderMissing)?;
```

Comment:
if we can't convert it to string ( i.e. it isn't utf8 ) shouldn't mean it was missing right? we can probably use the other variant which shows it wasn't "mqtt"?